packages/kbn-unified-data-table/src/components/data_table.scss

```diff
@@ -41,13 +41,17 @@
   }
 
   .euiDataGridHeaderCell {
-    align-items: start;
+    align-items: center;
```

Comment:
To align with the "graggable" state of the header column which is in a portal and we can't override it there.

Reply:
I think we can keep the original styles and have this work for dragging by adding this to our `EuiDataGridColumn` defs:
```ts
displayHeaderCellProps: { className: `unifiedDataTable__headerCell` }
```

Then we can remove these nested styles including `euiDataGridHeaderCell__popover` and `euiDataGridHeaderCell__draggableIcon`, and use a single global style:
```css
.euiDataGridHeaderCell.unifiedDataTable__headerCell {
  align-items: start;
}
```